File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/evaluation/DarknessEvaluator.java
package se.gustavkarlsson.aurora_notifier.android.evaluation;

import se.gustavkarlsson.aurora_notifier.android.models.factors.Darkness;

public class DarknessEvaluator implements ChanceEvaluator<Darkness> {
	public Chance evaluate(Darkness darkness) {
		Float zenithAngle = darkness.getSunZenithAngle();
		if (zenithAngle == null) {
			return Chance.unknown();
		}
		return Chance.of((1.0 / 12.0) * zenithAngle - 8.0); // 96-108
	}
}

File: app/src/main/java/se/gustavkarlsson/skylight/android/evaluation/DarknessEvaluator.kt
package se.gustavkarlsson.skylight.android.evaluation

import dagger.Reusable
import se.gustavkarlsson.skylight.android.evaluation.Chance.Companion.UNKNOWN
import se.gustavkarlsson.skylight.android.models.factors.Darkness
import javax.inject.Inject

@Reusable
internal class DarknessEvaluator
@Inject
constructor() : ChanceEvaluator<Darkness> {

    override fun evaluate(value: Darkness): Chance {
        val zenithAngle = value.sunZenithAngle ?: return UNKNOWN
        return Chance(1.0 / 12.0 * zenithAngle - 8.0) // 96-108
    }
}
